src/linechart.ts

```diff
@@ -1,89 +1,75 @@
-/// <reference path="../typings/angularjs/angular.d.ts" />
-/// <reference path="factories/Stage.ts" />
-
-var mod = angular.module('n3-line-chart', []);
-mod.directive('linechart', function() {
-  return {
-    restrict: 'E',
-    scope: {datasets: '=', options: '='},
-    link: function(scope:ng.IScope, element:JQuery) {
-      var stage: Stage = new Stage(element);
+/// <reference path='app.ts' />
+
+module n3Charts {
+  'use strict';
+
+  interface ILineChartScope extends ng.IScope {
+    data;
+    options;
+  }
+
+  export class LineChart implements ng.IDirective  {
+
+    public scope = {
+      datasets: '=',
+      options: '='
+    };
+
+    public restrict = 'E';
+    public replace = true;
+    public template = '<div></div>';
+
+    constructor(public em: Utils.EventManager, public fm: Utils.FactoryManager) {
```

Comment:
ooooh no :D I am just lazy
but you are right, nobody can guess the names from seeing this abbreviations